##DeepCE

## About
This directory contains the code and resources of the following paper:
"DeepCE: a deep learning framework for correlation-enhanced gene regulatory network inference in single-cell RNA sequencing data"

## Overview of the Model
The propose a deep learning framework DeepCE is illustrated in Figure 1. The framework comprises two main components: data processing and model construction. This design enables the model to jointly capture temporal dynamics and spatial patterns from the refined correlation matrices, facilitating accurate prediction of regulatory interactions.

<p align="center">
<img  src="model.png" width="800" height="400" > 
</p>

## Dependencies 
- python 3.7
- scanpy                        1.9.4
- numpy                         1.24.0
- pandas                        2.1.0
- scikit-learn                  1.3.0
- tensorflow-estimator          2.10.0
- tensorflow-gpu                2.10.0
- tensorflow-intel              2.14.0
- tensorflow-io-gcs-filesystem  0.26.0
- keras                         2.10.0

All dependencies can be installed within a few minutes.

##  Usage
DeepCE take data as input file in tsv, csv, 10X format, or h5ad format provided by Scanpy (genes in columns and cells in rows for tsv and csv).

Command to run DeepCE
```sh
python genepairlist_2.py
python matrix_3.py
python datasetSplit_4.py
python train_5.py
python predict_6.py
```
input

genepairlist_2.py:
gene_expression
gene_order
gold_network

python matrix_3.py：
gene_pair_list
gene_expression
gold_network

datasetSplit_4.py:
Calculate matrix generated by python matrix_3.py

python train_5.py：
Training set data


python predict_6.py：
test set data


output

genepairlist_2.py:
Generate a list of gene pairs with balanced positive and negative samples.
For each transcription factor (TF), matches each known positive TF-target interaction with one negative sample(a gene that is not its known target), ensuring a balanced dataset for supervised learning tasks.
If there are not enough negatives for a TF, negatives are borrowed from other TFs if available.
The resulting CSV file includes columns: ['TF', 'Target', 'Label', 'Score'],where Label is 1 for positive samples and 0 for negatives.

python matrix_3.py：
Calculate matrix for one gene pair based on sliding windows and lag correlations.


datasetSplit_4.py:
Divide the dataset in a ratio of 3:1:1.


train_5.py:
Train the model using the training set data.

predict_6.py:
This script performs model evaluation using a saved deep learning model.
It loads test data from `matrix_path`, evaluates model predictions on the data, and saves classification results, evaluation metrics, and plots (ROC, PR curves) to `save_dir`.
